.github/workflows/build.yml

```diff
@@ -352,7 +553,7 @@ jobs:
 
   upload-nightly:
     if: (github.ref == 'refs/heads/master') && (github.repository == 'airbus/scikit-decide') && (github.event_name == 'schedule')
-    needs: [test-unix, test-macos, test-windows]
+    needs: [build-docs, test-macos, test-windows]
```

Comment:
why this change?

Reply:
As explained in the description of the PR. Building the doc is done as early as possible. Since it is done on `ubuntu` it is done after tests have been performed. Therefore `build-docs` depends now from `test-linux` and `upload-nightly` depends on `build-docs`

Reply:
But upload-nightly has nothing to do with docs, so why does it depend to build-docs?

Reply:
ok went back to same structure